<launch>
  <!-- Object file Name without extension-->
  <arg name="object"         default="drill" />
  <arg name="data_path"      default="$(find sp_segmenter)/data" />
  <arg name="svm_path"       default="UR5_drill_svm" />
  <arg name="params"         default="" />
  <arg name="pairWidth"      default="0.05" />
  <arg name="NodeName"       default="SPServer" />
  <arg name="bestPoseOnly"   default="false" />
  <arg name="minConfidence"  default="0.18" doc="(float) confidence threshold for recognizing objects with ObjRecRANSAC" />
  <arg name="aboveTable"     default="0.01" />
  <arg name="tableTF"        default="camera_2/ar_marker_0" />
  <arg name="saveTabledir"   default="$(find sp_segmenter)/data" />
  <arg name="loadTable"      default="true" />
  <arg name="saveTable"      default="true" />

  <!-- SPSegmenterNode subscriber/publisher args -->
  <arg name="pcl_in"         default="/camera/depth_registered/points" />
  <arg name="pcl_out"        default="points_out"/>
  <arg name="poses_out"      default="poses_out"/>

  <node pkg="sp_segmenter" type="SPSegmenterServer" name="$(arg NodeName)" args="-p $(arg params)">  
    <!-- SPSegmenterNode Node arg pass -->
    <param name="cur_name"   type="str" value="$(arg object)" />
    <param name="bestPoseOnly" type="bool" value="$(arg bestPoseOnly)" />
    <param name="minConfidence"  type="double" value="$(arg minConfidence)"/>
    <param name="aboveTable"     type="double" value="$(arg aboveTable)"/>
    
    <param name="tableTF"        type="str" value="$(arg tableTF)"/>
    <param name="saveTable_directory"   type="str" value="$(arg saveTabledir)" />
    <param name="saveTable"   type="bool" value="$(arg saveTable)" />
    <param name="loadTable"   type="bool" value="$(arg loadTable)" />
    

    <!-- SPSegmenterNode subscriber/publisher args pass -->
    <param name="POINTS_IN"  type="str" value="$(arg pcl_in)"/>
    <param name="POINTS_OUT" type="str" value="$(arg pcl_out)"/>
    <param name="POSES_OUT"  type="str" value="$(arg poses_out)"/>
    <param name="pairWidth"  type="double" value="$(arg pairWidth)"/>
    <param name="svm_path"   type="str" value="$(arg data_path)/$(arg svm_path)/" />

    <!-- objectDatabase -->
    <rosparam command="load" file="$(find sp_segmenter)/launch/objectDatabase.yaml" />
  </node>

  <group ns="$(arg NodeName)">
    <include file="$(find sp_segmenter)/launch/parameters.launch">
        <arg name="data_path" value ="$(arg data_path)"/>
    </include>
  </group>
</launch>
